### PR TITLE
TAJO-1674: Validation of CTAS schema mismatch.

### PR DESCRIPTION
**Pull request by hyunsik**

### CODE REVIEW

**Comment by hyunsik** [comment]
Fixed test failures and rebased.

File: tajo-catalog/tajo-catalog-common/src/main/java/org/apache/tajo/catalog/CatalogUtil.java [hunk 1]
```diff
@@ -929,9 +933,16 @@ public static AlterTableDesc addOrDropPartition(String tableName, String[] colum
     TUtil.putToNestedMap(OPERATION_CASTING_MAP, Type.FLOAT8, Type.TEXT, Type.TEXT);
 
     TUtil.putToNestedMap(OPERATION_CASTING_MAP, Type.TEXT, Type.TIMESTAMP, Type.TIMESTAMP);
+    TUtil.putToNestedMap(OPERATION_CASTING_MAP, Type.TEXT, Type.TEXT, Type.TEXT);
```
Comment (blrunner) [inline tajo-catalog/tajo-catalog-common/src/main/java/org/apache/tajo/catalog/CatalogUtil.java]: We might need time type and date type.

File: tajo-core-tests/src/test/java/org/apache/tajo/engine/query/TestTablePartitions.java [hunk 2]
```diff
@@ -211,7 +208,7 @@ public final void testCreateColumnPartitionedTableWithSelectedColumns() throws E
 
     TableDesc tableDesc = catalog.getTableDesc(DEFAULT_DATABASE_NAME, tableName);
     verifyPartitionDirectoryFromCatalog(DEFAULT_DATABASE_NAME, tableName, new String[]{"key"},
-      tableDesc.getStats().getNumRows());
+        tableDesc.getStats().getNumRows());
```
Comment (blrunner) [inline tajo-core-tests/src/test/java/org/apache/tajo/engine/query/TestTablePartitions.java]: This class seems to have unnecessary modifications.

**Comment by hyunsik** [comment]
I reflected the comments and rebased.

**Comment by jihoonson** [comment]
+1 ship it!

**Pull request closed without merging**